include/pybind11/cast.h

```diff
@@ -36,6 +36,15 @@ class type_caster : public type_caster_base<type> {};
 template <typename type>
 using make_caster = type_caster<intrinsic_t<type>>;
 
+template <typename T>
+struct is_generic_type<T, enable_if_t<std::is_base_of<type_caster_generic, make_caster<T>>::value>>
```

Comment:
From @rwgk in #2950

> Two questions:
> 
> 1. Will this work if I add another specialization pair for `is_base_of<smart_holder_type_caster_base_tag, ...>`?
> 
> 2. What do you think about choosing a different name, to connect to the intended purpose instead of pegging the name on `type_caster_generic`, e.g. `is_iterable_t_compatible`? (Otherwise it will get really confusing that generic is connected to `smart_holder_type_caster`.)

(1) Yup! Can simply add the specialization and it should work out of the box.
(2) I know that was for prior PR; I believe this trait is more agnostic. See PR overview for rationale.

Reply:
What is the public API use case? Versus leaving inside of a detail namespace?